### ISSUE TITLE
Adding a neural network for images capture too

### ISSUE THREAD

**Issue by abozb01** [abozb01/Intelligent-Customer-Support#1]
import numpy as np
import matplotlib.pyplot as plt
from sklearn.datasets import make_blobs
from keras.preprocessing.image import ImageDataGenerator
from keras.models import Sequential
from keras.layers import Conv2D, MaxPooling2D, Flatten, Dense

# Generate random data points
X, y = make_blobs(n_samples=100, centers=2, random_state=42)

# Reshape data for CNN input
X = X.reshape(-1, 1, 2, 1)

# Data augmentation with random cropping
datagen = ImageDataGenerator(width_shift_range=[-2, 2], height_shift_range=[-2, 2])
datagen.fit(X)

# Visualize data before data augmentation
plt.figure(figsize=(12, 6))
for i in range(6):
    plt.subplot(2, 3, i+1)
    plt.scatter(X[i, 0, :, 0], X[i, 0, :, 0], c=y[i])
plt.suptitle('Data before Data Augmentation', fontsize=16)
plt.show()

# Create a simple CNN model
model = Sequential()
model.add(Conv2D(32, kernel_size=(3, 3), activation='relu', input_shape=(1, 2, 1)))
model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Flatten())
model.add(Dense(1, activation='sigmoid'))

# Compile the model
model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])

# Visualize data after data augmentation
plt.figure(figsize=(12, 6))
for X_batch, y_batch in datagen.flow(X, y, batch_size=6):
    for i in range(6):
        plt.subplot(2, 3, i+1)
        plt.scatter(X_batch[i, 0, :, 0], X_batch[i, 0, :, 0], c=y_batch[i])
    break  # Break the loop after one batch
plt.suptitle('Data after Data Augmentation', fontsize=16)
plt.show()

# Train the model with augmented data
model.fit(datagen.flow(X, y, batch_size=6), steps_per_epoch=len(X) // 6, epochs=5)

# Make predictions with the model
# Add your prediction code here

# Visualize CNN results
# Add code to visualize the CNN results
